File: 0844-backspace-string-compare/0844-backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        stack1 = []
        stack2 = []

        for ch in s:
            if ch != '#':
                stack1.append(ch)
            elif stack1:
                stack1.pop()
        for ch in t:
            if ch != '#':
                stack2.append(ch)
            elif stack2:
                stack2.pop()
        
        return stack1 == stack2

File: 0844-backspace-string-compare/0844-backspace-string-compare.c
#include <stdbool.h>
#include <string.h>

bool backspaceCompare(char* s, char* t) {

    char stack1[205];
    char stack2[205];

    int top1 = -1;
    int top2 = -1;

    // Process first string
    for(int i = 0; s[i] != '\0'; i++) {

        if(s[i] != '#') {
            stack1[++top1] = s[i];
        }
        else if(top1 >= 0) {
            top1--;
        }
    }

    // Process second string
    for(int i = 0; t[i] != '\0'; i++) {

        if(t[i] != '#') {
            stack2[++top2] = t[i];
        }
        else if(top2 >= 0) {
            top2--;
        }
    }

    // Different lengths means different strings
    if(top1 != top2)
        return false;

    // Compare stacks
    for(int i = 0; i <= top1; i++) {

        if(stack1[i] != stack2[i])
            return false;
    }

    return true;
}